src/pages/levelpack/Records.js

```diff
@@ -28,80 +33,180 @@ const Records = ({
   records,
   recordsLoading,
   showLegacyIcon,
+  showMoreStats,
+  levelStats,
 }) => {
   const [level, selectLevel] = useState(-1);
   const [longName, setLongName] = useState('');
   const [levelName, setLevelName] = useState('');
 
+  const { setShowMoreStats } = useStoreActions(state => state.LevelPack);
+
   if (recordsLoading) {
     return <Loading />;
   }
 
+  // eslint-disable-next-line no-unused-vars
+  const anyAreLegacy = hasSource(records);
+
+  const MaxRelativeTimeAll = Math.max(
+    ...Object.values(levelStats || {}).map(v => v.RelativeTimeAll),
+  );
+
   return (
     <>
-      <Header h2 mLeft>
-        Levels
-      </Header>
+      <Row ai="center" jc="space-between" width="100%">
+        <Header h2 mLeft>
+          Levels
+        </Header>
+        <Switch checked={showMoreStats} onChange={setShowMoreStats}>
+          Level Stats
+        </Switch>
+      </Row>
+
       <ListContainer>
         <ListHeader>
-          <ListCell width={100}>Filename</ListCell>
-          <ListCell width={320}>Level name</ListCell>
-          <ListCell width={200}>Kuski</ListCell>
-          <ListCell>Time</ListCell>
-          {hasSource(records) && <ListCell />}
-        </ListHeader>
-        {records.map(r => (
-          <TimeRow
-            key={r.LevelIndex}
-            onClick={e => {
-              e.preventDefault();
-              selectLevel(level === r.LevelIndex ? -1 : r.LevelIndex);
-              setLongName(r.Level.LongName);
-              setLevelName(r.Level.LevelName);
-            }}
-            selected={level === r.LevelIndex}
-          >
-            <ListCell width={100}>
-              <Level LevelIndex={r.LevelIndex} LevelData={r.Level} />
+          <ListCell width={90}>Filename</ListCell>
+          <ListCell width={showMoreStats ? 260 : 320}>Level Name</ListCell>
+          <ListCell width={150}>Kuski</ListCell>
+          <ListCell width={150}>Time</ListCell>
+
+          {showMoreStats && <ListCell width={170}>Last Driven</ListCell>}
+
+          {showMoreStats && (
+            <ListCell width={140}>
+              Kuski's Played <br />
+              <span title="Percentage of kuski's that finished the level at least once.">
+                (% Finished)
+              </span>
             </ListCell>
-            <ListCell width={320}>{r.Level.LongName}</ListCell>
-            {r.LevelBesttime ? (
-              <>
-                <ListCell width={200}>
+          )}
+
+          {showMoreStats && (
+            <ListCell
+              width={50}
+              title="Total time played by all kuski's combined."
+            >
+              Time Played
+            </ListCell>
+          )}
+
+          <ListCell
+            width={150}
+            title="Time played on level (by all kuski's), relative to other levels in this pack."
+          >
+            Popularity
+          </ListCell>
+
+          {!showMoreStats && <ListCell />}
+        </ListHeader>
+        {records.map(r => {
```

Comment:
maybe refactor the insides of the map to own component, don't like having so much logic in the middle of the html

Reply:
I moved into own component. Not sure it's worth it because I had to pass like 12 props to it but parent component is shorter now so that's fine I guess. I also accidentally did the same for kuski/Records.js at first. i'll probably just leave that now that i've already changed it.